ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+#
+#
+# This module implement a rule-based editor that is used to modify the content
+# of an existing file without taking care of the whole file. Each rule
+# driving the edition process is applied to one matching line. The input for
+# updating the file is the Quattor configuration and conditions can be defined
+# based on the contents of this configuration.
+#
+#######################################################################
+
+package NCM::Component::DPMLFC::RuleBasedEditor;
+
+use strict;
+use NCM::Component;
+use vars qw(@ISA $EC);
+@ISA = qw(NCM::Component);
+$EC=LC::Exception::Context->new->will_store_all;
+
+use EDG::WP4::CCM::Element;
+
+use Readonly;
+
+use File::Path;
+use File::Copy;
+use File::Compare;
+use File::Basename;
+use File::stat;
+
+use LC::Check;
+use CAF::FileWriter;
+use CAF::FileEditor;
+use CAF::FileReader;
+use CAF::Process;
+use CAF::Object;
+
+use Encode qw(encode_utf8);
+use Fcntl qw(SEEK_SET);
+
+local(*DTA);
+
+
+# Constants use to format lines in configuration files
+# Keep in sync with ncm-xrootd
+use constant LINE_FORMAT_PARAM => 1;
+use constant LINE_FORMAT_ENVVAR => 2;
+use constant LINE_FORMAT_XRDCFG => 3;
+use constant LINE_FORMAT_XRDCFG_SETENV => 4;
+use constant LINE_FORMAT_XRDCFG_SET => 5;
+use constant LINE_VALUE_AS_IS => 0;
+use constant LINE_VALUE_BOOLEAN => 1;
+use constant LINE_VALUE_HOST_LIST => 2;
+use constant LINE_VALUE_INSTANCE_PARAMS => 3;
+use constant LINE_VALUE_ARRAY => 4;
+use constant LINE_VALUE_HASH_KEYS => 5;
+use constant LINE_VALUE_STRING_HASH => 6;
+use constant LINE_VALUE_OPT_NONE => 0;
+use constant LINE_VALUE_OPT_SINGLE => 1;
+use constant LINE_FORMAT_DEFAULT => LINE_FORMAT_PARAM;
+use constant LINE_QUATTOR_COMMENT => "\t\t# Line generated by Quattor";
+use constant LINE_OPT_DEF_REMOVE_IF_UNDEF => 0;
+use constant LINE_OPT_DEF_ALWAYS_RULES_ONLY => 0;
+
+
+# Define and export the constants used to define rules
+# Keep in sync with ncm-xrootd
+# See http://www.perlmonks.org/?node_id=1072731 for the exporting method
+#use warnings;
+#my %rule_constants;
+#BEGIN {
+#  %rule_constants = (
+#    "LINE_FORMAT_PARAM" => 1,
+#    "LINE_FORMAT_ENVVAR" => 2,
+#    "LINE_FORMAT_XRDCFG" => 3,
+#    "LINE_FORMAT_XRDCFG_SETENV" => 4,
+#    "LINE_FORMAT_XRDCFG_SET" => 5,
+#    "LINE_VALUE_AS_IS" => 0,
+#    "LINE_VALUE_BOOLEAN" => 1,
+#    "LINE_VALUE_HOST_LIST" => 2,
+#    "LINE_VALUE_INSTANCE_PARAMS" => 3,
+#    "LINE_VALUE_ARRAY" => 4,
+#    "LINE_VALUE_HASH_KEYS" => 5,
+#    "LINE_VALUE_STRING_HASH" => 6,
+#    "LINE_VALUE_OPT_NONE" => 0,
+#    "LINE_VALUE_OPT_SINGLE" => 1,
+#    "LINE_FORMAT_DEFAULT" => LINE_FORMAT_PARAM,
+#    "LINE_QUATTOR_COMMENT" => "\t\t# Line generated by Quattor",
+#    "LINE_OPT_DEF_REMOVE_IF_UNDEF" => 0,
+#    "LINE_OPT_DEF_ALWAYS_RULES_ONLY" => 0,
+#  );
+#}
+
+#use constant \%rule_constants;
+
+use base 'Exporter';
```

Comment:
this is a bit dangerous, mixing ISA and base. better `use parent qw(NCM::Component Exporter);`